util/src/rate_counter.rs

```diff
@@ -103,8 +104,8 @@ impl RateCounter {
 // turns out getting the millisecs since epoch in Rust isn't as easy as it
 // could be
 fn millis_since_epoch() -> u64 {
-	let since_epoch = SystemTime::now()
-		.duration_since(SystemTime::UNIX_EPOCH)
-		.unwrap_or(Duration::new(0, 0));
-	since_epoch.as_secs() * 1000 + u64::from(since_epoch.subsec_millis())
+	match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+		Ok(since_epoch) => since_epoch.as_millis().try_into().unwrap_or(0),
+		Err(_) => 0,
```

Comment:
`.ok_or(0)` ?

Reply:
Good point. Addressed it and potentially created a monster 👹.

Reply:
Looks like this is a bit to cutting edge for the CI. `map_or` was introduced in Rust 1.41.

Reply:
Can be replaced by:
```
// turns out getting the millisecs since epoch in Rust isn't as easy as it
// could be
fn millis_since_epoch() -> u64 {
	SystemTime::now()
		.duration_since(SystemTime::UNIX_EPOCH)
		.map(|since_epoch| since_epoch.as_millis().try_into().unwrap_or(0))
		.unwrap_or(0)
}
```